### ISSUE TITLE
Add support for other web servers

### ISSUE THREAD

**Issue by matjazpotocnik** [processwire/processwire-requests#7]
At https://processwire.com/docs/install/new/ in the Requirements section is written "A web server running Apache.". I think it would be good to mention that PW can also run on other web servers/platforms, although not officially supported.

**Comment by teppokoivula** [comment]
I would suggest taking this one step further and adding official support for other web servers. I'm aware of multiple cases where ProcessWire has been used successfully on nginx and IIS, so we're probably not too far from there.

**Comment by matjazpotocnik** [comment]
See https://github.com/ryancramerdesign/ProcessWire/issues/1453

**Comment by clsource** [comment]
also runs on caddy
https://github.com/caddyserver/examples/tree/master/processwire

**Comment by ryancramerdesign** [comment]
I'd be comfortable updating the language to say "Apache or 100% compatible". I know that PW will run on plenty of other servers, but 100% compatibility with our .htaccess file is pretty darn important for security. So what I'm not comfortable with is saying we're compatible with servers that I can't vouch for in terms of the security issues. If we've got experts that can monitor and vouch for these things on specific server platforms, and maintain whatever changes are necessary to the .htaccess file (or whatever file the server uses) then that would be okay. nginx seems to be the one that comes up most often.

**Comment by matjazpotocnik** [comment]
> I'd be comfortable updating the language to say "Apache or 100% compatible".

Hm, it's better than nothing but I'm not sure those are the right words. I think the only web server that is 100% compatible with Apache is Apache. Don't get me wrong, I understand your concern about security, so am I. When I discovered PW, the section about security was one of the first chapters I read. More than once. Before I put our site live, I checked multiple scenarios in attempt to find security hole and failed. Are there any tests we can perform to test our web config file?

**Comment by teppokoivula** [comment]
>  I think the only web server that is 100% compatible with Apache is Apache.

Agreed. Don't want to shoot down the whole idea, but this would just be just a roundabout way of saying "compatible with Apache only". I'm also pretty sure that this would raise more questions than answer, so my suggestion is to not go with such wording.

(Based on my experience so far even systems that aim for Apache compatibility (such as LiteSpeed) are actually far from being "100% compatible". Even if they claim so, and I'm pretty sure they don't.)

I've been diving a bit into nginx lately and supporting it officially would, in my opinion, be relatively painless. I'll get back to that topic soon, hopefully. In the meantime I would suggest mentioning the "community ports" for other web servers, but including a strict warning: UNOFFICIAL, NOT RECOMMENDED FOR PRODUCTION USE, USE AT YOUR OWN RISK, etc.

Recommending an unofficial port and just dropping a slight hint that it exists are two very different things. While the first one would be preferable and could potentially widen our user base noticeably, the former would already be an improvement to current "Apache only" situation.

**Comment by ivangretsky** [comment]
I think that the wording does not need to be short. Why not put a paragraph or two describing the steps to be taken to provide other web servers support? If someone is going for nginx he or she is probably aware of the complexity it might bring. 

Example: "Nginx or about any other web server is supported if it can provide the means to reproduce the configuration of a .htaccess file shiped in the core package. For nginx go to [https://winginx.com/en/htaccess](https://winginx.com/en/htaccess) or similar service".

**Comment by teppokoivula** [comment]
> Recommending an unofficial port and just dropping a slight hint that it exists are two very different things. While the first one would be preferable and could potentially widen our user base noticeably, the former would already be an improvement over current "Apache only" situation.

One thing I forgot to mention there: in my opinion mentioning some of these "community features" right within the "official docs" would actually be a good thing, as it would send a clear message about ProcessWire having an open and active community behind it.

In that regard it would be similar to the docs [mentioning the Multisite module](https://processwire.com/api/modules/multi-site-support/).

**Comment by teppokoivula** [comment]
@ivangretsky: to be honest I'd prefer not directing users to automated services. Admittedly I'm not aware of the one you mentioned and it could actually do downright amazing job, but I would still prefer pointing to a version that has been checked by a human :)

**Comment by LostKobrakai** [comment]
I feel so, too. Linking to selected community efforts might focus any collaboration attempts on a complete solution. I mean there's a probably 95% complete nginx config burried in the forum. If that one would get some attention it could probably be kept in sync with the .htaccess by the people using nginx. I'm certain there are a few people out there using their own nginx configs, which could iron out any potentially missing peaces.

**Comment by ivangretsky** [comment]
The service is just an example. Of course a link to a community supported config is much better. The thing is that there is no need for complete core supported solution to mention compatibility with some warnings.

**Comment by dlzi** [comment]
I completely agree with @LostKobrakai suggestion. So, a first step would be to create a section like "Other Servers Support" on the forum, and keep the respective configs there till PW officially support them. Is that possible?

**Comment by matjazpotocnik** [comment]
@ryancramerdesign, thank you for the updated Requirements section! I find it very good.

> A Unix or Windows-based web server running Apache (and others, see next section)

Where is that next section? I'm reading this link: https://processwire.com/docs/install/new/
Ok, I see what's going on: you are using Profields Text Blocks to pull text from this link: https://processwire.com/about/requirements/ and there is next section. Maybe you should remove the words "see next section"?

Next, maybe you should change this:

> For instance, we are aware of several instances using nginx, IIS and other web server software.

to this:

> For instance, we are aware of several instances using nginx, IIS, caddy and possibly other web server software.

because we know PW works on caddy. Maybe you could mention that MariaDB is as good as MySQL?

> Multi-byte string support compiled with PHP (mbstring)

That should be required. Strange results can occur when mb_\* functions are not used where it should have been. Also, Punycode, Parsedown and ParsedownExtra don't check for mbstring support.

**Comment by matjazpotocnik** [comment]
Maybe we should add curl and zip PHP extensions to the list of requirements? I tried Soma’s Online Installer and it failed because I didn't have zip extension installed.
